internal/stack/stack_processor.go

```diff
@@ -39,29 +53,38 @@ func ProcessYAMLConfigFiles(filePaths []string) ([]string, error) {
 // ProcessYAMLConfigFile takes a path to a YAML config file,
 // recursively processes and deep-merges all imports,
 // and returns stack config as map[interface{}]interface{}
-func ProcessYAMLConfigFile(filePath string) (map[interface{}]interface{}, error) {
+func ProcessYAMLConfigFile(filePath string, importsList *[]string) (map[interface{}]interface{}, *[]string, error) {
```

Comment:
Do you really want a pointer here? I would consider passing by value unless there is a strong reason. It makes unexpected side-effects less likely.

Reply:
it's uses recursion, and by-value did not work